compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -254,8 +255,17 @@ private class ExtractAPICollector(implicit val ctx: Context) extends ThunkHolder
     val cinfo = csym.classInfo
 
     val bases =
-      try linearizedAncestorTypes(cinfo)
-      catch {
+      try {
+        val ancestorTypes0 = linearizedAncestorTypes(cinfo)
+        if (ValueClasses.isDerivedValueClass(csym)) {
+          val underlying = ValueClasses.valueClassUnbox(csym).info.finalResultType
+          // The underlying type of a value class should be part of the name hash
+          // of the value class (see the test `value-class-underlying`), this is accomplished
+          // by adding the underlying type to the list of parent types.
+          underlying :: ancestorTypes0
+        } else
+          ancestorTypes0
+      } catch {
```

Comment:
The try/catch should only be needed around the call to ` linearizedAncestorTypes(cinfo)`.